dask/dataframe/core.py

```diff
@@ -2415,6 +2415,14 @@ def to_series(self):
         return self.map_partitions(M.to_series,
                                    meta=self._meta.to_series())
 
+    @derived_from(pd.Index)
+    def to_frame(self, name=None):
+        if PANDAS_VERSION < '0.24.0' and name is not None:
+            raise ValueError('renaming index is not supported on this PANDAS version')
```

Comment:
Adding information about what the current pandas version is and when `name=` was added could be useful when this error message is raised. For example, something like:

```python
raise ValueError("The 'name' keyword was added in pandas 0.24.0. "
                 "Your version of pandas is '{}'.".format(PANDAS_VERSION))
```